README.md
[![CRAN_Status_Badge](http://www.r-pkg.org/badges/version/SightabilityModel)](https://cran.r-project.org/package=SightabilityModel)

[![](https://cranlogs.r-pkg.org/badges/SightabilityModel)](https://cran.r-project.org/package=SightabilityModel)

[![](http://cranlogs.r-pkg.org/badges/grand-total/SightabilityModel?color=orange)](https://cran.r-project.org/package=SightabilityModel)

# SightabilityModel
Contains code for the SightabilityModel Package

## Versions and installation

  * **Github** To install the latest development version from Github, 
    install the newest version of the **devtools** package; then run
```
devtools::install_github("jfieberg/SightabilityModel", dependencies = TRUE,
                        build_vignettes = TRUE)
```
## Features

Estimates abundance, density, and ratio of variables in simple random 
sample or stratified random sample block design often used in studying
ungulate populations. Can also be used with Domain Stratification (Heard et al., 2008)
but SE will have to be estimated using bootstrapping.

In a typical study, the survey area is divided into strata and each
stratum divided into blocks. Not all blocks need be the same area.
Aerial surveys of selected blocks find groups of animals and the number
of animals in the group is recorded, often divided by sex (bulls and cows)
and age (calve and mature).

A key problem is that not all animals in a group are seen
because sightability is < 1. A separate study of a known
number of animal (e.g. radio collared) is used to estimate
a sightability model. The actual sightability study data or the fitted
mode from a sightability model using logistic regression is used to
correct the observed counts in the study. The corrected (for sightability)
counts are used to estimate density, abundance or ratios (e.g. bulls to cows).

## References

Fieberg, J. 2012. Estimating Population Abundance Using Sightability Models: R SightabilityModel Package. Journal of Statistical Software, 51(9), 1-20. URL https://doi.org/10.18637/jss.v051.i09

Heard, D.C. A.B.D. Walker, J.B. Ayotte, and G.S. Watts. 2008. 
Using GIS to modify a stratified random block survey design for moose. 
Alces 44: 11-116.

Steinhorst, Kirk R. and Samuel, Michael D. 1989. Sightability Adjustment Methods for Aerial Surveys of Wildlife Populations. Biometrics 45:415–425.

Wong, C. 1996. Population size estimation using the modified Horvitz-Thompson estimator with estimated sighting probabilities. Dissertation, Colorado State University, Fort Collins, USA.



NEWS.md
# SightabilityModel 1.5.4

* Correct documentation as requested by CRAN. NO changes to functionality.

# SightabilityModel 1.5.3

* Substantial Improvement in computational time for Wong.est() function.

# SightabilityModel 1.5.2

* Resubmission to remove Imports/Suggests: tidyverse
* Typographical errors corrected
* No new functionality.

# SightabilityModel 1.5.1

* Fixup vignette index to sort properly
* Misc small changes to vignettes
* No new functionality added.

# SightabilityModel 1.5

* Added new vignette dealing with Domain stratification (Heard et al 2008)
* Fixed MoosePopR() and SightabilityPopR() to deal with 100% census
* Added new functions
  + compute.SCF() to compute sightability function given data and model
  + MoosePopR_DomStrat() for domain stratification. Note that SE have not been
  adjusted for multiple domains measured on the same sampling unit. Bootstrapping may 
  be required.
  + SightabilityPopR_DomStrat() with similar functionality for sightability models
  + MoosePopR_DomStrat_bootrep() generates a bootstrap replicate and analysis. Allows
  for adjustments for sightability. Can be used for sightability models (mean-per-unit estimates) as well by
  setting block area to 1 and total block area to number of blocks.
* Numerous typographical corrections in code and vignettes

# SightabilityModel 1.4.2

* Updated URLs to Journal of Statistical Software to DOI as recommended by JSS editors.
* No change in functionality





R/.-package.R


#' Experimental (test trials) data set used to estimate detection probabilities
#' for moose in MN
#' 
#' Experimental (test trials) data set used to estimate detection probabilities
#' for moose in MN
#' 
#' 
#' @name exp.m
#' @docType data
#' @format A data frame with 124 observations on the following 4 variables.
#' \describe{ \item{list("year")}{year of the experimental survey (test trial)}
#' \item{list("observed")}{Boolean variable (=1 if moose was observed and 0
#' otherwise)} \item{list("voc")}{measurement of visual obstruction}
#' \item{list("grpsize")}{group size (number of observed moose in each
#' independently sighted group)} }
#' @references Giudice, J H. and Fieberg, J. and Lenarz, M. S.  2012. Spending
#' Degrees of Freedom in a Poor Economy: A Case Study of Building a
#' Sightability Model for Moose in Northeastern Minnesota.  Journal of Wildlife
#' Management 76(1):75-87.
#' @keywords datasets
#' @examples
#' 
#' data(exp.m)
#' exp.m[1:5,]
#' 
NULL





#' Mountain Goat Sightabilty Model Information
#' 
#' Model averaged regression parameters and unconditional variance-covariance
#' matrix for mountain goat sightability model (Rice et al. 2009)
#' 
#' 
#' @name g.fit
#' @docType data
#' @format The format is: beta.g = list of regression parameters (intercept and
#' parameters associated with GroupSize, Terrain, and X.VegCover) varbeta.g =
#' variance-covariance matrix (associated with beta.g)
#' @references Rice C.G., Jenkins K.J., Chang W.Y. (2009). A Sightability Model
#' for Mountain Goats. The Journal of Wildlife Management, 73(3), 468-478.
#' @keywords datasets
#' @examples
#' 
#' data(g.fit)
#' 
NULL





#' Mountain Goat Survey Data from Olympic National park
#' 
#' Mountain Goat Survey Data from Olympic National park collected in 2004
#' 
#' 
#' @name gdat
#' @docType data
#' @format A data frame with 113 observations on the following 9 variables.
#' \describe{ \item{list("GroupSize")}{number of animals observed in each
#' independently sighted group [cluster size]} \item{list("Terrain")}{measure
#' of terrain obstruction} \item{list("pct.VegCover")}{measure of vegetative
#' obstruction} \item{list("stratum")}{stratum identifier}
#' \item{list("total")}{number of animals observed in each independently
#' sighted group [same as GroupSize]} \item{list("subunit")}{a numeric vector,
#' Plot ID} }
#' @references Jenkins, K. J., Happe, P.J., Beirne, K.F, Hoffman, R.A.,
#' Griffin, P.C., Baccus, W. T., and J. Fieberg.  In press.  Recent population
#' trends in mountain goats in the Olympic mountains.  Northwest Science.
#' @source Patti Happe (Patti_Happe@nps.gov)
#' @keywords datasets
#' @examples
#' 
#' data(gdat)
#' 
NULL





#' MN moose survey data
#' 
#' Operational survey data for moose in MN (during years 2004-2007).  Each
#' record corresponds to an indpendently sighted group of moose, with variables
#' that capture individual covariates (used in the detection model) as well as
#' plot-level information (stratum identifer, sampling probability, etc).
#' 
#' 
#' @name obs.m
#' @docType data
#' @format A data frame with 805 observations on the following 11 variables.
#' \describe{ \item{list("year")}{year of survey}
#' \item{list("stratum")}{stratum identifier} \item{list("subunit")}{sample
#' plot ID} \item{list("total")}{number of moose observed}
#' \item{list("cows")}{number of cows observed} \item{list("calves")}{number of
#' calves observed} \item{list("bulls")}{number of bulls observed}
#' \item{list("unclass")}{number of unclassified animals observed (could not
#' identify sex/age class)} \item{list("voc")}{ measurement of visual
#' obstruction} \item{list("grpsize")}{group size (cluter size)} }
#' @references Giudice, J H. and Fieberg, J. and Lenarz, M. S.  2012. Spending
#' Degrees of Freedom in a Poor Economy: A Case Study of Building a
#' Sightability Model for Moose in Northeastern Minnesota.  Journal of Wildlife
#' Management 76(1):75-87.
#' @keywords datasets
#' @examples
#' 
#' data(obs.m) 
#' obs.m[1:5, ]
#' 
NULL





#' Data set containing sampling information for observation survey of moose in
#' MN
#' 
#' Data set containing sampling information from a survey of moose in MN
#' (during years 2004-2007)
#' 
#' 
#' @name sampinfo.m
#' @docType data
#' @format A data frame with 12 observations on the following 5 variables.
#' \describe{ \item{list("year")}{year of survey}
#' \item{list("stratum")}{stratum identifier} \item{list("Nh")}{number of
#' population units in stratum h} \item{list("nh")}{number of sample units in
#' stratum h} }
#' @references Giudice, J H. and Fieberg, J. and Lenarz, M. S.  2012. Spending
#' Degrees of Freedom in a Poor Economy: A Case Study of Building a
#' Sightability Model for Moose in Northeastern Minnesota.  Journal of Wildlife
#' Management 76(1):75-87.
#' @keywords datasets
#' @examples
#' 
#' data(sampinfo.m)
#' sampinfo.m
#' 
NULL





#' Wildlife Sightability Modeling
#' 
#' Uses logistic regression to model the probability of detection as a function
#' of covariates.  This model is then used with observational survey data to
#' estimate population size, while accounting for uncertain detection.  See
#' Steinhorst and Samuel (1989).
#' 
#' \tabular{ll}{ Package: \tab SightabilityModel\cr Type: \tab Package\cr
#' Version: \tab 1.3\cr Date: \tab 2014-10-03\cr License: \tab GPL-2\cr
#' LazyLoad: \tab yes\cr }
#' 
#' @name SightabilityModel-package
#' @aliases SightabilityModel-package SightabilityModel
#' @docType package
#' @author John Fieberg
#' 
#' Maintainer: John Fieberg <jfieberg@@umn.edu>
#' @references Fieberg, J.  2012.  Estimating Population Abundance Using
#' Sightability Models: R SightabilityModel Package. Journal of Statistical
#' Software, 51(9), 1-20.  URL http://www.jstatsoft.org/v51/i09/.
#' 
#' Steinhorst, Kirk R. and Samuel, Michael D. 1989. Sightability Adjustment
#' Methods for Aerial Surveys of Wildlife Populations. Biometrics 45:415--425.
#' @keywords package
NULL



